ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -3086,6 +3086,13 @@ Seems much cleaner if each stmt is identified as a particular HiveOperation (whi
             output.getWriteType().name()));
         break;
 
+      case CTAS:
+        if (AcidUtils.isExclusiveCTASEnabled(conf) && AcidUtils.isTransactionalTable(t)) {
```

Comment:
Could we move `AcidUtils.isTransactionalTable(t)` check inside of isExclusiveCTASEnabled?

Reply:
No, since we do not have table data at the other places where the isExclusiveCTASEnabled method is called.